composer.lock

```diff
@@ -4672,37 +4750,38 @@
             "description": "A small library for converting tokenized PHP source code into XML and potentially other formats",
             "support": {
                 "issues": "https://github.com/theseer/tokenizer/issues",
-                "source": "https://github.com/theseer/tokenizer/tree/master"
+                "source": "https://github.com/theseer/tokenizer/tree/1.2.1"
             },
             "funding": [
                 {
                     "url": "https://github.com/theseer",
                     "type": "github"
                 }
             ],
-            "time": "2020-07-12T23:59:07+00:00"
+            "time": "2021-07-28T10:34:58+00:00"
         },
         {
             "name": "vimeo/psalm",
-            "version": "4.6.4",
+            "version": "4.10.0",
```

Comment:
You've updated *everything* and thus also updated `vimeo/psalm` which now leads to psalm errors.
Thats why you should only use `composer update --lock` if you change something in the `composer.json` which is not actually installing anything (like config stuff, non-installable dependencies like extensions, php versions, e.g.).

If you have to bump packages to be compatible with PHP 8.1, you might want to require these new package versions explicitly.

You can also bump psalm but then fixing the errors should be mandatory.  If you don't want to fix them, do not update psalm.

Reply:
Totally overlooked that. Reverted accidental dependencies upgrade.